src/components/RouteForDeliveryModal/RouteForDeliveryModal.test.js

```diff
@@ -0,0 +1,171 @@
+        value: defaultProps.isPrintableByDefault,
+      };
+
+      expect(Checkbox).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+
+    it('Appropriate label of "Checkbox" component should be rendered', () => {
+      expect(screen.getByText(messageIds.printSlip)).toBeInTheDocument();
+    });
+
+    it('"Close and checkout" button should be called with correct props', () => {
+      const expectedProps = {
+        buttonStyle: 'primary',
+        onClick: defaultProps.onCloseAndCheckout,
+      };
+
+      expect(Button).toHaveBeenNthCalledWith(buttonCallOrder.closeAndCheckout, expect.objectContaining(expectedProps), {});
+    });
+
+    it('"Close" button should be called with correct props', () => {
+      const expectedProps = {
+        buttonStyle: 'primary',
+        onClick: defaultProps.onClose,
+      };
+
+      expect(Button).toHaveBeenNthCalledWith(buttonCallOrder.close, expect.objectContaining(expectedProps), {});
+    });
+
+    it('"Close and checkout" button label should be rendered in the document', () => {
+      expect(screen.getByText(messageIds.closeAndCheckout)).toBeInTheDocument();
+    });
+
+    it('"Close" button label should be rendered in the document', () => {
+      expect(screen.getByText(messageIds.close)).toBeInTheDocument();
+    });
+  });
+
+  describe('When "isPrintableByDefault" true and "slipTemplate" not defined', () => {
+    const props = {
+      ...defaultProps,
+      isPrintableByDefault: true,
+      slipTemplate: undefined,
+    };
+
+    beforeEach(() => {
+      render(
+        <RouteForDeliveryModal
+          {...props}
+        />
+      );
+    });
+
+    it('"Close and checkout" print button should be called with correct props', () => {
+      const expectedProps = {
+        buttonStyle: 'primary',
+        onBeforePrint: defaultProps.onCloseAndCheckout,
+        dataSource: defaultProps.slipData,
+        template: '',
+      };
+
+      expect(PrintButton).toHaveBeenNthCalledWith(buttonCallOrder.closeAndCheckout, expect.objectContaining(expectedProps), {});
+    });
+
+    it('"Close" print button should be called with correct props', () => {
+      const expectedProps = {
+        buttonStyle: 'primary',
+        onBeforePrint: defaultProps.onClose,
+        dataSource: defaultProps.slipData,
+        template: '',
+      };
+
+      expect(PrintButton).toHaveBeenNthCalledWith(buttonCallOrder.close, expect.objectContaining(expectedProps), {});
+    });
+
+    it('"Close and checkout" print button label should be rendered in the document', () => {
+      expect(screen.getByText(messageIds.closeAndCheckout)).toBeInTheDocument();
+    });
+
+    it('"Close" print button label should be rendered in the document', () => {
+      expect(screen.getByText(messageIds.close)).toBeInTheDocument();
+    });
+
+    it('After clicking on checkbox "PrintButton" should disappear', () => {
+      const printButtons = screen.getAllByTestId('print-button');
```

Comment:
Do we really have multiple  print buttons?

Reply:
And does it really works? As far as I remember we should use queryByTestId if element is not in the DOM. If we use getByTestId instead then it will throw exception

Reply:
> Do we really have multiple print buttons?

Yes, we have 2 PrintButtons at the same time (see screenshot below).

![image](https://user-images.githubusercontent.com/42437614/163402588-c215e61d-5962-43dc-9d5d-d2c6f2637570.png)

Reply:
> 

Yes, it works.
My approach works because I try to find buttons, then click on Checkbox and buttons should disappear. 
If I initially click on Checkbox and then try to find buttons I will have error.

But your suggestion also should work correctly.

Reply:
> And does it really works? As far as I remember we should use queryByTestId if element is not in the DOM. If we use getByTestId instead then it will throw exception

Yes, it works.
My approach works because I try to find buttons, then click on Checkbox and buttons should disappear.
If I initially click on Checkbox and then try to find buttons I will have error.

But your suggestion also should work correctly.

Reply:
I see. Thanks!

Reply:
But anyway I would add different testid for each button and would looks for required button